api/src/main/java/org/apache/iceberg/expressions/ExpressionUtil.java

```diff
@@ -258,6 +270,23 @@ public <T> String predicate(UnboundPredicate<T> pred) {
     }
   }
 
+  private static <T> List<String> abbreviateValues(List<String> sanitizedValues) {
+    if (sanitizedValues.size() >= LONG_IN_PREDICATE_ABBREVIATION_THRESHOLD) {
+      Set<String> distinctValues = ImmutableSet.copyOf(sanitizedValues);
+      if (distinctValues.size()
+          <= sanitizedValues.size() - LONG_IN_PREDICATE_ABBREVIATION_MIN_GAIN) {
+        List<String> abbreviatedList = Lists.newArrayListWithCapacity(distinctValues.size() + 1);
+        abbreviatedList.addAll(distinctValues);
+        abbreviatedList.add(
+            String.format(
+                "... (%d values hidden, %d in total) ...",
+                sanitizedValues.size() - distinctValues.size(), sanitizedValues.size()));
+        return abbreviatedList;
+      }
+    }
+    return sanitizedValues;
```

Comment:
In Iceberg, we add an empty newline between a control flow block and the following statement.